crates/templates/src/run.rs

```diff
@@ -248,6 +249,18 @@ impl Run {
                         Err(anyhow::anyhow!("Spin doesn't know what to do with a 'component' snippet outside an 'add component' operation")),
                 }
             },
+            "variables" => {
+                match &self.options.variant {
+                    TemplateVariantInfo::AddComponent { manifest_path } =>
+                        Ok(RenderOperation::MergeToml(
+                            manifest_path.clone(),
+                            MergeTarget::Application("variables"),
+                            content,
+                        )),
+                    TemplateVariantInfo::NewApplication =>
```

Comment:
Could we add support for new templates with variables in a future PR?

Reply:
New templates can have variables - I'm not sure what you're getting at here.

Reply:
I was missing the deeper dive into the code base of how we are defining the snippets  only under the AddComponent variant. I now see how this path isn't executed during adds. Thanks!